presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -378,7 +378,7 @@ public static Boolean stIsClosed(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(BOOLEAN)
     public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        return deserialize(input).isEmpty();
+        return deserializeEnvelope(input) == null;
```

Comment:
It should be 

```
Envelope envelope = deserializeEnvelope(input) ;
return envelope == null || envelope.isEmpty();
```

As for the `Envelope` geometry type empty envelope is returned instead of `null`

Reply:
@arhimondr This is an interesting observation. I wanted to add a test that breaks with current implementation, but couldn't. Turns out that we never serialize empty envelopes:

```
public static Slice serialize(Envelope envelope)
    {
        requireNonNull(envelope, "envelope is null");
        verify(!envelope.isEmpty());
```

Reply:
It may change in the future, and than the `stIsEmpty` will be returning invalid results.